mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -3861,6 +3876,28 @@ public DocumentationMember (XmlNode node)
 			for (int i = 0; i < p.Count; ++i)
 				Parameters.Add (p [i].Attributes ["Type"].Value);
 		}
+		XmlNodeList tp = node.SelectNodes ("TypeParameters/TypeParameter[not(@apistyle) or @apistyle='classic']");
+		if (tp.Count > 0) {
+			TypeParameters = new StringList (tp.Count);
+			for (int i = 0; i < tp.Count; ++i)
+				TypeParameters.Add (tp [i].Attributes ["Name"].Value);
+		}
+		else {
+			DiscernTypeParameters ();
```

Comment:
Unfortunately, this won't always be the case, particularly for importing docs generated from triple-slash ... for example:
https://github.com/mono/mono/blob/master/mcs/tools/mdoc/Test/TestEcmaDocs.xml#L317

The docs there don't have a `TypeParameter` list, so I went with the approach of checking for `TypeParameters`, and otherwise falling back to checking for type parameters in the name string.